src/chat-model.ts

```diff
@@ -623,23 +623,39 @@ ${toolsList}
     }
 
     try {
-      const model = await this.input.documentManager?.services.contents.get(
-        attachment.value
-      );
-      if (!model || model.type !== 'notebook') {
-        return null;
-      }
+      // Try reading from live notebook if open
+      const widget = this.input.documentManager?.findWidget(attachment.value);
```

Comment:
As far as I know, the widget can only be a `IDocumentWidget<Notebook, INotebookModel> | undefined`. 
Casting it would avoid typing the following `sharedModel` and `cells` as `any`.
Did you see use case where the widget is of another type ?